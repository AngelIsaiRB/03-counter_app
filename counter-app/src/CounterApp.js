import React, {useState} from 'react';
import PropTypes from "prop-types";


const CounterApp = ({value}) =>{

    const [counter, setCounter ]=useState(value);



    // handleAndd
    const handleAdd=(e)=>{
        setCounter(counter+e);
        // setCounter((c)=>c+1);
    }
    

    const handleReset=()=>{
        setCounter(0);
    }

    return <> 
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            {/* <button onClick={()=>handleAdd()}</button>}> */}
            <button onClick={()=>handleAdd(1)}>+1</button>
            <button  onClick={()=>handleAdd(-1)}>-1</button>
            <button onClick={handleReset}>0</button>
            </>;
};

CounterApp.propTypes={
    value: PropTypes.number,
}

CounterApp.defaultProps = {
    value: 10,

}

export default CounterApp;

