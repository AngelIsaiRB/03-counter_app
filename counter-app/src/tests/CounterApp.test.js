import React from 'react';
import  { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('pruebas en el <CounterApp/>', () => {
    
    test('should de regresar el valor predefinido de 10', () => {
        const wrapper =  shallow(<CounterApp />); 
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should de regresar el valor predefinido de 10', () => {
        const valueI=100;
        const wrapper = shallow(<CounterApp value={valueI}/>);
        const textValor = wrapper.find("h2").text().trim();
        expect(textValor).toBe(`${valueI}`);
        
    });
    

});
