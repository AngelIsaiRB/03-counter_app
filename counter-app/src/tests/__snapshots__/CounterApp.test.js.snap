// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <CounterApp/> should de regresar el valor predefinido de 10 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    0
  </button>
</Fragment>
`;
